### PR TITLE
Use placeholder dimensions that reflect the final text layout

### PR DESCRIPTION
**Pull request by jason-simmons**
The _computeChildren[Width/Height]With[Min/Max]Intrinsics methods in
RenderParagraph overwrite the text painter's placeholder dimensions
in order to compute potential widths and heights for the paragraph.

The placeholder dimensions should be restored to the values computed by
_layoutChildren before painting the text.

### CODE REVIEW

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -488,7 +488,10 @@ class RenderParagraph extends RenderBox
     _textPainter.markNeedsLayout();
   }
 
+  List<PlaceholderDimensions> _placeholderDimensions;
```
Comment (GaryQian) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Might be worth adding non-public documentation explaining why this is needed.

**Comment by GaryQian** [comment]
Needs goldens hash update.

**Review by GaryQian — APPROVED** [review]
LGMT! Thanks for fixing this!

File: packages/flutter/lib/src/rendering/paragraph.dart (hunk 1 above)
Comment (jason-simmons) [reply packages/flutter/lib/src/rendering/paragraph.dart]: done

**Comment by jason-simmons** [comment]
Goldens update: https://github.com/flutter/goldens/pull/57

**Pull request merged**